### PR TITLE
httpd.c: cleanup funky version detection/reporting

### PR DESCRIPTION
**Pull request by ksmurchison**

### CODE REVIEW

File: imap/httpd.c [hunk 1]
```diff
@@ -1383,6 +1383,11 @@ static int reset_saslconn(sasl_conn_t **conn)
 }
 
 
+static const char *http_versions[] = {
+    "HTTP/1.0", HTTP_VERSION, HTTP2_VERSION
+};
+
```
Comment (elliefm) [inline imap/httpd.c:1389]: If you add something like this here...

```suggestion
static const char *http_versions[] = {
    "HTTP/1.0", HTTP_VERSION, HTTP2_VERSION
};
static const size_t n_http_versions = sizeof(http_versions) / sizeof(http_versions[0]);
```

File: imap/httpd.c [hunk 2]
```diff
@@ -2595,12 +2597,7 @@ EXPORTED const char *http_statusline(unsigned ver, long code)
 {
     static struct buf statline = BUF_INITIALIZER;
 
-    if (ver == VER_2) buf_setcstr(&statline, HTTP2_VERSION);
-    else {
-        buf_setmap(&statline, HTTP_VERSION, HTTP_VERSION_LEN-1);
-        buf_putc(&statline, ver + '0');
-    }
-
+    buf_setcstr(&statline, http_versions[ver]);
```
Comment (elliefm) [inline imap/httpd.c:2600]: ... then you could add an assertion here for safety:

```suggestion
    assert(ver < n_http_versions);
    buf_setcstr(&statline, http_versions[ver]);
```

**Review by elliefm — APPROVED** [review]
Looks good, one suggestion if you like

File: imap/httpd.c (hunk 1 above)
Comment (rsto) [reply imap/httpd.c:1389]: In addition, you might want to change
`static const char *http_versions[]`
to
`static const char * const http_versions[]`

**Review by rsto — APPROVED** [review]

File: imap/httpd.c (hunk 1 above)
Comment (ksmurchison) [reply imap/httpd.c]: done

File: imap/httpd.c (hunk 2 above)
Comment (ksmurchison) [reply imap/httpd.c:2604]: done

**Pull request merged**